File: application/backend-monorepo-multiservice/code/src/Common/Event/Event.php
<?php

declare(strict_types=1);

namespace Galeas\Api\Common\Event;

use Galeas\Api\Common\Id\Id;

interface Event
{
    public function eventId(): Id;

    public function aggregateId(): Id;

    public function aggregateVersion(): int;

    public function causationId(): Id;

    public function correlationId(): Id;

    public function recordedOn(): \DateTimeImmutable;

    public function metadata(): array;
}

File: application/backend-credit-card-enrollment/backend-c#/Common/Event/Event.cs
namespace CreditCardEnrollment.Common.Event;

public abstract class Event
{
    public required string EventId { get; init; }
    public required string AggregateId { get; init; }
    public required int AggregateVersion { get; init; }
    public required string CorrelationId { get; init; }
    public required string CausationId { get;  init; }
    public required DateTime RecordedOn { get; init; }
}
